bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -305,57 +305,77 @@ public static Type getConvertibleType(Object inputValue, Type targetType, String
                 return TypeConverter.resolveMatchingTypeForUnion(inputValue, targetType);
             case TypeTags.INTERSECTION_TAG:
                 Type effectiveType = ((BIntersectionType) targetType).getEffectiveType();
-                return isFromJson ? getConvertibleTypeFromJson(inputValue, effectiveType, varName,
-                        unresolvedValues, errors, allowNumericConversion) : getConvertibleType(inputValue,
-                        effectiveType, varName, false, unresolvedValues, errors, allowNumericConversion);
+                return getConvertibleType(inputValue, effectiveType, varName, unresolvedValues, errors,
+                        allowNumericConversion);
             case TypeTags.FINITE_TYPE_TAG:
-                return getConvertibleFiniteType(inputValue, (BFiniteType) targetType, varName, isFromJson,
+                return getConvertibleFiniteType(inputValue, (BFiniteType) targetType, varName,
                         errors, unresolvedValues, allowNumericConversion);
             case TypeTags.TYPE_REFERENCED_TYPE_TAG:
                 return getConvertibleType(inputValue, ((BTypeReferenceType) targetType).getReferredType(), varName,
-                        isFromJson, unresolvedValues, errors, allowNumericConversion);
+                        unresolvedValues, errors, allowNumericConversion);
             case TypeTags.TYPEDESC_TAG:
                 return getConvertibleType(inputValue, ((BTypedescType) targetType).getConstraint(), varName,
-                        isFromJson, unresolvedValues, errors, allowNumericConversion);
+                        unresolvedValues, errors, allowNumericConversion);
             default:
-                if (TypeChecker.checkIsLikeType(inputValue, targetType, allowNumericConversion)) {
+                if (TypeChecker.checkIsLikeType(inputValue, targetType, allowNumericConversion)
+                        || (TypeTags.isXMLTypeTag(targetTypeTag) && isStringConvertibleToXmlType(inputValue))) {
                     return targetType;
                 }
         }
         return null;
     }
 
+    private static boolean isStringConvertibleToXmlType(Object inputValue) {
+        if (TypeChecker.getType(inputValue).getTag() == TypeTags.STRING_TAG) {
+            try {
+                XmlFactory.parse(((BString) inputValue).getValue());
+            } catch (BError e) {
+                return false;
```

Comment:
What would be the error message printed down the line in case the parsing results in a `BError`? Shouldn't we show the actual cause of the error in conversion?

Reply:
This method returns a boolean which shows whether the provided value is an string that can be parsed as an xml. If it cannot be parsed the `XmlFactory.parse` method will throw an BError. Here we do not need to show the actual cause of the error.